apps/desktop/src/lib/ai/service.ts

```diff
@@ -209,16 +212,19 @@ export class AIService {
 	// This optionally returns a summarizer. There are a few conditions for how this may occur
 	// Firstly, if the user has opted to use the GB API and isn't logged in, it will return undefined
 	// Secondly, if the user has opted to bring their own key but hasn't provided one, it will return undefined
-	async buildClient(userToken?: string): Promise<Result<AIClient, Error>> {
+	async buildClient(): Promise<Result<AIClient, Error>> {
 		const modelKind = await this.getModelKind();
 
 		if (await this.usingGitButlerAPI()) {
-			if (!userToken) {
+			// TODO(CTO): Once @estib has landed the new auth, it would be good to
+			// about a good way of checking whether the user is authenticated.
+			if (!get(this.tokenMemoryService.token)) {
 				return buildFailureFromAny(
 					"When using GitButler's API to summarize code, you must be logged in"
 				);
 			}
-			return ok(new ButlerAIClient(this.cloud, userToken, modelKind));
+
+			return ok(new ButlerAIClient(this.cloud, modelKind));
```

Comment:
It would be nice to have something on a shared UserService that we can call to determine whether the user is authenticated, or even what level of permissions the user has, but that didn't fit into the scope of this PR